### PR TITLE
Sym upstream PR 3 Inverse PINN Example

### PR DESCRIPTION
**Pull request by ktangsali**
# PhysicsNeMo Pull Request

## Description
Adds an inverse PINN example for the Sym Upstream effort

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/cfd/inverse_pinns/train_inverse_pinn.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+            ),
+        }
+
+
+def _download_data(csv_file: str, url: str, log: PythonLogger) -> None:
+    """Download CSV if it does not already exist."""
+    if not os.path.exists(csv_file):
+        log.info(f"Downloading {csv_file} ...")
+        urllib.request.urlretrieve(url, csv_file)
+    else:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/inverse_pinns/train_inverse_pinn.py:97]: This is a known URL so it should not have an issue.

File: examples/cfd/inverse_pinns/train_inverse_pinn.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+    # Training loop
+    max_steps = cfg.training.max_steps
+    log_freq = cfg.training.log_freq
+    batch_size = cfg.training.batch_size
+
+    for step in range(max_steps):
+        optimizer.zero_grad()
+
+        idx = np.random.choice(len(coords), size=batch_size, replace=False)
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/inverse_pinns/train_inverse_pinn.py:211]: The README discusses this, and the traceback would be meaningful. I don't want to add unnecessary complexity to the code.

File: examples/cfd/inverse_pinns/train_inverse_pinn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/inverse_pinns/train_inverse_pinn.py:97]: Yes, example script and we want to keep it minimal to focus on the core of inverse problem. Fixing this requires a lot of extra code to the _download_data function.

File: physicsnemo/sym/eq/phy_informer.py [hunk 3]
```diff
@@ -59,6 +59,18 @@ class PhysicsInformer:
     compute_connectivity : bool
         If True and using ``"least_squares"``, build the connectivity tensor
         on the fly from ``"nodes"`` and ``"edges"`` in the input dict.
+    detach_names : list[str] or None
```
Comment (loliverhennigh) [inline physicsnemo/sym/eq/phy_informer.py:62]: This changes the API slightly but I think its ok. Like if someone was already using this they might get accidentally pass device for the detach_names.

**Review by loliverhennigh — APPROVED** [review]
LGTM

File: physicsnemo/sym/eq/phy_informer.py (hunk 3 above)
Comment (ktangsali) [reply physicsnemo/sym/eq/phy_informer.py:62]: Yes, fair point, but this `PhysicsInformer` was upstreamed recently, the probability of that happening is kinda low :). Eitherways, this is needed to support the inverse type of problems :)
